app/assets/javascripts/supplemental_files.js

```diff
@@ -0,0 +1,40 @@
+
+$('button[name="edit_label"]').on('click', e => {
+  const { $row, fileId, masterFileId } = getHTMLInfo(e);
+  const inputField = $row.find('input[name="label_' + masterFileId + '_' + fileId + '"]');
+  inputField.val($row.data('file-label'));
+  $row.addClass('is-editing');
+  inputField.focus();
+});
+
+$('button[name="label_edit_cancel"]').on('click', e => {
+  const { $row, fileId, masterFileId } = getHTMLInfo(e);
+  $row.find('input[name="label_' + masterFileId + '_' + fileId + '"]');
+  $row.removeClass('is-editing');
+  });
+
+$('button[name="save_label"]').on('click', e => {
+  const {  $row, fileId, masterFileId } = getHTMLInfo(e);
+  const newLabel = $row.find('input[name="label_' + masterFileId + '_' + fileId + '"]').val();
+  $row.find('span[name="label_' + masterFileId + '_' + fileId + '"]').text(newLabel);
+
+  let formData = new FormData();
+  formData.append('supplemental_file[label]', newLabel);
+  formData.append('authenticity_token', $('input[name=authenticity_token]').val());
+
+  fetch('/master_files/' + masterFileId + '/supplemental_files/' + fileId, {
+    method: "PUT",
+    body: formData
+  }).then(() => {
+    $row.removeClass('is-editing');
+    // Page reload to show the flash message
+    location.reload();
```

Comment:
I wonder if it is really necessary to reload the page for a label update.  Maybe an inline indication of success would be better?  But I'd wait on feedback from PO before changing this since it looks like it already works.

Reply:
Yeah, that sounds good to me. Now that I think about it, some inline edit forms do not have feedback flash messages (e.g. editing markers in playlists, add to playlist in item page. these are the inline forms I can think of right now).